eng/pipelines/templates/jobs/ci.tests.yml

```diff
@@ -43,10 +43,23 @@ jobs:
       - ${{ if not(contains(variables['Build.DefinitionName'], '-pr - ')) }}:
         - template: /eng/common/pipelines/templates/steps/sparse-checkout.yml
           parameters:
+            ${{ if eq(parameters.ProjectListOverrideFilePropertyName, '') }}:
+              Paths:
+                - "/*"
+                - "!SessionRecords"
+                - "/sdk/${{ parameters.ServiceDirectory }}/**/SessionRecords"
+            ${{ else }}:
               Paths:
                 - "/*"
                 - "!SessionRecords"
-                - "/sdk/${{ parameters.ServiceToTest }}/**/SessionRecords"
+                - "/sdk/*/**/SessionRecords"
+      - ${{ if ne(parameters.ProjectListOverrideFilePropertyName, '') }}:
+        - task: DownloadPipelineArtifact@2
+          inputs:
+            artifact: DependencyTestProjectReferences
+            patterns: "**/$(${{ parameters.ProjectListOverrideFilePropertyName }})"
```

Comment:
Is the `**/` necessary here?

Reply:
Let me try. The folder is like 
```
DependencyTestProjectReferences\Project_0.props
DependencyTestProjectReferences\Project_1.props
...
```

Reply:
OK if it is under a sub-directory this is probably needed.